### DISCUSSION TITLE
Is there a way to receive notices or messages immediately upon establishing a connection?

### DISCUSSION THREAD

**Discussion by akshay-joshi** [psycopg/psycopg#952] [General]
I have a scenario where a user’s password is nearing expiration. A warning or hint message is displayed when the user connects to EDB’s Advanced Server via PSQL. I added the **add_notice_handler** function, but I cannot capture these notices since the connection has already been established when they appear.

<img width="831" alt="Screenshot 2024-11-11 at 2 56 50 PM" src="https://github.com/user-attachments/assets/5860a329-cc08-474b-8d4e-62f5186f2430">

Any pointers/suggestions are welcome, we need to show this in pgAdmin 4.

**Comment by dvarrazzo** [comment]
I assume it will take some refactoring, likely to set up the handler before connection. I also don't have a way to test the change or to develop the feature unless a similar communication pattern can be reproduced on a freely available server.

I am open to receive reasonable merge requests to support the feature (where "reasonable" means breaking no current test, possibly adding new ones exercising the feature, adding documentation), but, because it is about supporting a non-free vendor, I am not available to give development time for free. EDB is welcome to sponsor the project.

**Comment by akshay-joshi** [comment]
Thanks, Daniele

We will check if can provide the merge request.
